pkg/storage/storage.go

```diff
@@ -0,0 +1,191 @@
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/google/uuid"
+	_ "github.com/mattn/go-sqlite3"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/carfloresf/the-sun-god/config"
+	"github.com/carfloresf/the-sun-god/internal/model"
+)
+
+const (
+	insertPostQuery = `INSERT
+						INTO post(id,title,author,subreddit,link,content,score,promoted,nsfw) 
+						VALUES    ($1, $2, $3, $4, $5, $6, $7, $8, $9)`
+	updatePostScoreQuery = `UPDATE post SET score = $1 WHERE id = $2`
+	countTotalPostsQuery = `SELECT COUNT(*) FROM post WHERE promoted = false`
+	queryPosts           = `SELECT id,title,author,subreddit,link,content,score,promoted,nsfw,created_at 
+							FROM post WHERE promoted = $1 ORDER BY score DESC, created_at DESC LIMIT $2 OFFSET $3;`
+	queryPromotedPosts = `SELECT id,title,author,subreddit,link,content,score,promoted,nsfw,created_at 
+							FROM post WHERE promoted = true ORDER BY RANDOM() LIMIT $1;`
+)
+
+var allStatements = []string{insertPostQuery, updatePostScoreQuery, countTotalPostsQuery, queryPosts, queryPromotedPosts}
+
+type DB struct {
+	Conn               *sql.DB
+	preparedStatements map[string]*sql.Stmt
+}
+
+func NewStorage(config *config.DB) (*DB, error) {
+	sqliteDatabase, err := sql.Open("sqlite3", config.DBFile)
+	if err != nil {
+		log.Fatal("error opening Conn connection: %w", err)
+		return nil, err
+	}
+
+	db := DB{
+		Conn: sqliteDatabase,
+	}
+
+	return &db, nil
+}
+
+func (d *DB) PrepareAllStatements() error {
```

Comment:
My understanding is that this should only be called on startup, so you can maybe use sync.Once to ensure you are not preparing statements multiple times

Also, maybe it can be good to add new function like: `NewStorageWithPreparedStatements`. The PrepareAllStatements is kind of requirement to be called because all of the logic relies on prepared statements being populated. It seems better from the API perspective for programmers not to have to know about the logic you are relying, and in a way, they have to know the PrepareAllStatements is a requirement for this package to function properly when db conn is established.